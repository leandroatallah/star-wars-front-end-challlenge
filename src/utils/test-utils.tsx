import { Provider } from 'react-redux'
import { render } from '@testing-library/react'
import { CHANGE_THEME } from '../config'
import { createStore, Store } from 'redux'

const startingState = { theme: '' }
function reducer(state = startingState, action: { type: string; payload: string }) {
    switch (action.type) {
        case CHANGE_THEME:
            return { ...state, theme: action.payload }
        default:
            return state
    }
}

type storeProps = {
    initialState?: {
        theme: string
    },
    store?: Store<{
        theme: string;
    }, {
        type: string;
        payload: string;
    }>
}

export function renderWithRedux(
    component: JSX.Element,
    { initialState, store = createStore(reducer, initialState) }: storeProps = {},
) {
    return {
        ...render(<Provider store={store} > {component} </Provider>),
    }
}
