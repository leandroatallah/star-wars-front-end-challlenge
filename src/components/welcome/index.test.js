import '@testing-library/jest-dom'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'
import Welcome from '.'
import themeColors from '../../config/theme-colors'
import { renderWithRedux } from '../../utils/test-utils'

describe('<Welcome />', () => {
    it('Render with redux', () => {
        const { getByTestId } = renderWithRedux(
            <ThemeProvider theme={themeColors}>
                <Welcome />
            </ThemeProvider>,
        )
        expect(getByTestId('welcome')).toMatchSnapshot()
    })
})
