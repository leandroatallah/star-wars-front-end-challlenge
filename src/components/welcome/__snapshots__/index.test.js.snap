// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Welcome /> Render with redux 1`] = `
.c1 {
  background-color: #2C97D1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 56px;
  font-size: 18px;
  line-height: 1.218;
  font-weight: 700;
  padding: 0 37px;
  border-radius: 10px;
  color: #FFF;
  cursor: pointer;
  pointer-events: auto;
}

.c1 svg {
  display: none;
  width: 20px;
  height: 20px;
  margin: 0;
  margin-left: 10px;
  vertical-align: middle;
}

.c0 {
  margin-top: 210px;
  text-align: center;
  color: #2C97D1;
}

.c0 h1 {
  font-size: 72px;
  line-height: 1.2;
  margin-bottom: 7px;
  font-weight: 400;
}

.c0 h1 strong {
  font-weight: 700;
}

.c0 h2 {
  font-size: 14px;
  line-height: 1.218;
  font-weight: 400;
  -webkit-letter-spacing: calc(14px * 0.35);
  -moz-letter-spacing: calc(14px * 0.35);
  -ms-letter-spacing: calc(14px * 0.35);
  letter-spacing: calc(14px * 0.35);
  text-transform: uppercase;
  margin-bottom: 162px;
}

<section
  class="c0"
  data-testid="welcome"
>
  <h1>
    Welcome to 
    <strong>
      iClinic
    </strong>
  </h1>
  <h2>
    Frontend Challenge
  </h2>
  <button
    class="c1"
    data-testid="button"
    loading="0"
    type="button"
  >
    <span>
      S T A R T
    </span>
    <svg
      preserveAspectRatio="xMidYMid"
      viewBox="0 0 100 100"
      xmlns="http://www.w3.org/2000/svg"
      xmlns:xlink="http://www.w3.org/1999/xlink"
    >
      <circle
        cx="50"
        cy="50"
        fill="none"
        r="35"
        stroke="#ffffff"
        stroke-dasharray="164.93361431346415 56.97787143782138"
        stroke-width="10"
      >
        <animatetransform
          attributeName="transform"
          dur="1s"
          keyTimes="0;1"
          repeatCount="indefinite"
          type="rotate"
          values="0 50 50;360 50 50"
        />
      </circle>
    </svg>
  </button>
</section>
`;
